``None`` is getting cast into a string in ``ProcessResult``

```
    def test_instance_types():                                                                            
        ret = ProcessResult(returncode=0, stdout="STDOUT", stderr="STDERR")                                                                                                                                         
        assert isinstance(ret.stdout, str)                                                                                                                                                                          
        assert isinstance(ret.stderr, str)                                                                                                                                                                          
        ret = ProcessResult(returncode=0, stdout=None, stderr=None)                                       
>       assert ret.stdout is None                                                                         
E       AssertionError: assert 'None' is None                                                             
E        +  where 'None' = ProcessResult(returncode=0, stdout='None', stderr='None', cmdline=None, data_key=None, data=None).stdout
                                                                                                          
ret        = ProcessResult(returncode=0, stdout='None', stderr='None', cmdline=None, data_key=None, data=None)
                                                                                                                                                                                                                    
tests/unit/utils/processes/test_processresult_matcher.py:29: AssertionError
```

--- a/changelog/4.bugfix.rst
+++ b/changelog/4.bugfix.rst
@@ -1,0 +1,1 @@
+Stop casting ``None`` to a string for ``ProcessResult.std{out,err}``

--- a/src/pytestshellutils/downgraded/utils/processes.py
+++ b/src/pytestshellutils/downgraded/utils/processes.py
@@ -37,6 +37,15 @@ class MatchString(str):
         return LineMatcher(self.splitlines())
 
 
+def convert_string_to_match_string(value: Optional[str]) -> Optional[MatchString]:
+    """
+    Convert strings into ``MatchString`` instances.
+    """
+    if isinstance(value, str):
+        return MatchString(value)
+    return value
+
+
 @attr.s(frozen=True, kw_only=True)
 class ProcessResult:
     """
@@ -66,8 +75,8 @@ class ProcessResult:
     """
 
     returncode = attr.ib()
-    stdout = attr.ib(converter=MatchString)
-    stderr = attr.ib(converter=MatchString)
+    stdout = attr.ib(converter=convert_string_to_match_string)
+    stderr = attr.ib(converter=convert_string_to_match_string)
     cmdline = attr.ib(default=None)
     data_key = attr.ib(default=None)
     data = attr.ib()

--- a/src/pytestshellutils/utils/processes.py
+++ b/src/pytestshellutils/utils/processes.py
@@ -39,6 +39,15 @@ class MatchString(str):
         return LineMatcher(self.splitlines())
 
 
+def convert_string_to_match_string(value: Optional[str]) -> Optional[MatchString]:
+    """
+    Convert strings into ``MatchString`` instances.
+    """
+    if isinstance(value, str):
+        return MatchString(value)
+    return value
+
+
 @attr.s(frozen=True, kw_only=True)
 class ProcessResult:
     """
@@ -68,8 +77,8 @@ class ProcessResult:
     """
 
     returncode: int = attr.ib()
-    stdout: str = attr.ib(converter=MatchString)
-    stderr: str = attr.ib(converter=MatchString)
+    stdout: str = attr.ib(converter=convert_string_to_match_string)
+    stderr: str = attr.ib(converter=convert_string_to_match_string)
     cmdline: Optional[List[str]] = attr.ib(default=None)
     data_key: Optional[str] = attr.ib(default=None)
     data: Optional[Dict[Any, Any]] = attr.ib()

--- a/tests/unit/utils/processes/test_processresult_matcher.py
+++ b/tests/unit/utils/processes/test_processresult_matcher.py
@@ -25,6 +25,12 @@ def test_instance_types():
     ret = ProcessResult(returncode=0, stdout="STDOUT", stderr="STDERR")
     assert isinstance(ret.stdout, str)
     assert isinstance(ret.stderr, str)
+    ret = ProcessResult(returncode=0, stdout=None, stderr=None)
+    assert ret.stdout is None
+    assert ret.stderr is None
+    ret = ProcessResult(returncode=0, stdout=1, stderr=2, data=None)
+    assert ret.stdout == 1
+    assert ret.stderr == 2
 
 
 def test_matcher_attribute(process_result):

